app/models/setting.rb

```diff
@@ -70,6 +72,25 @@ def generate_override_code!(size = 6)
     self.save!
   end
 
+  def generate_incoming_sms_token(replace=false)
+    # Don't replace token unless replace==true
+    unless self.incoming_sms_token.nil? or replace
+      return
+    end
+
+    # Ensure that the new token is actually different
+    begin
+      new_token = SecureRandom.hex
+    end while new_token == self.incoming_sms_token
+
+    self.incoming_sms_token = new_token
+  end
```

Comment:
Our styleguide says to only use `self` when required. So please remove these if you would. Thanks.